src/details/ArborX_Callbacks.hpp

```diff
@@ -179,32 +179,32 @@ void check_valid_callback(Callback const &callback, Predicates const &)
   using PredicateTag = typename AccessTraitsHelper<Access>::tag;
   using Predicate = typename AccessTraitsHelper<Access>::type;
 
-  static_assert(
-      (std::is_same<PredicateTag, SpatialPredicateTag>{} ||
-       std::is_same<PredicateTag, NearestPredicateTag>{}) &&
-          Kokkos::is_detected<Experimental_CallbackArchetypeExpression,
-                              Callback, Predicate, int>{},
-      "Callback 'operator()' does not have the correct signature");
+  static_assert(is_valid_predicate_tag<PredicateTag>::value,
+                "The predicate tag is not valid");
 
-  static_assert(
-      (std::is_same<PredicateTag, SpatialPredicateTag>{} &&
-       (std::is_same<
-            CallbackTreeTraversalControl,
-            Kokkos::detected_t<Experimental_CallbackArchetypeExpression,
-                               Callback, Predicate, int>>{} ||
-        std::is_void<
-            Kokkos::detected_t<Experimental_CallbackArchetypeExpression,
-                               Callback, Predicate, int>>{})) ||
-          std::is_same<PredicateTag, NearestPredicateTag>{},
-      "Callback 'operator()' return type must be void or "
-      "ArborX::CallbackTreeTraversalControl");
+  static_assert(Kokkos::is_detected<Experimental_CallbackArchetypeExpression,
+                                    Callback, Predicate, int>{},
+                "Callback 'operator()' does not have the correct signature");
 
   static_assert(
-      std::is_same<PredicateTag, SpatialPredicateTag>{} ||
-          (std::is_same<PredicateTag, NearestPredicateTag>{} &&
+      !(std::is_same<PredicateTag, SpatialPredicateTag>{} ||
+        std::is_same<PredicateTag,
+                     Experimental::OrderedSpatialPredicateTag>{}) ||
```

Comment:
Shouldn't this be 
```suggestion
      (std::is_same<PredicateTag, SpatialPredicateTag>{} ||
       std::is_same<PredicateTag,
                    Experimental::OrderedSpatialPredicateTag>{}) &&
```

Reply:
This says as much as if the predicate is `spatial` or `ordered_spatial`, the return type must be `void` or `CallbackTreeTraversalControl`.
The next one says: if the predicate is `nearest`, the return type must be `void`.

I think that I didn't change what we check (except for adding the new predicate, of course) but made both `static_asserts` more self-contained (in that we don't mention other predicates than the ones we want to impose a condition on).